### DISCUSSION TITLE
Updating AutoColumn resize to work with SleekGrid

### DISCUSSION THREAD

**Discussion by DGoertzen** [serenity-is/Serenity#6802] [Help]
Hello,

I tried implementing "Grid Slickgrid AutoColumn resize" from https://github.com/serenity-is/Serenity/wiki/Grid-Slickgrid-AutoColumn-resize.
When I implemented as instructed, I get the error "Error TS2694 (TS) Namespace 'global.Slick' has no exported member 'Grid'.

My assumption is that this is because of the change from Slickgrid to Sleekgrid as documented here: https://github.com/serenity-is/sleekgrid

1) Can somebody tell me if my understanding is correct that the error above is because of the change from to Slickgrid to Sleekgrid?
2) If the problem I am observing is from Slickgrid to Sleekgrid, has anybody already got "AutoColumn resize" to work with Sleekgrid?
3) If it not a problem from the change from to Slickgrid to Sleekgrid, any idea what I might be doing wrong?

Thanks!
Doug

**Comment by Surenthiran** [comment]
[silk.autoresize.js.txt](https://github.com/serenity-is/Serenity/files/11845175/silk.autoresize.js.txt)

put this file in path Scripts/Slick/slick.autocolumnsize.js 

create file slick.autocolumnsize.d.ts with following code in helper folder 

```
declare namespace Slick {
    declare function AutoColumnSize(initialAutosize: Boolean): void;
}
export as namespace Slick
```

you can resize any grid with following code 

`import { Grid } from '@serenity-is/sleekgrid';`

add this block after constructor in any grid 
```

    private autoSize;
    protected createSlickGrid(): Grid {
        let grid = super.createSlickGrid();
        this.autoSize = new Slick.AutoColumnSize(true); // *** If you want to make it autoresize when grid is loaded, use: new Slick.AutoColumnSize(true);
        grid.registerPlugin(this.autoSize);

        return grid;
    }

    // *** Resizes the columns after changes are made to the grid (ex. Column Picker) *
    protected markupReady() {
        super.markupReady();
        if (this.autoSize != null) {
            this.autoSize.resizeAllColumns();
        }
    }
```
